### PR TITLE
002 work log entry

### PR DESCRIPTION
**Pull request by totodo713**
## Summary

Work log entry機能の実装PRです。CSV入出力、代理入力機能、メンバー管理などを含みます。

### Backend
- **Event Sourcing**: Work log entry のイベントソーシング実装
- **CSV Import/Export**: CSVファイルのインポート・エクスポート機能
- **Member Repository**: メンバー管理とマネージャー階層のJDBC実装
- **Proxy Entry**: 代理入力のための認可チェック機能

### Frontend
- **CSV Uploader**: ドラッグ&ドロップ対応のCSVアップロードコンポーネント
- **API Service**: CSRF対応のAPIクライアント
- **Import Page**: CSVインポートページ

### Commit `12c7786` - Initial review fixes

**Cookie Parsing Fix**
- `api.ts`, `csvService.ts`: `split("=")` を `indexOf("=")` ベースの分割に変更。Base64エンコードされたCSRFトークンの `=` パディング文字を正しく処理

**CsvUploader Improvements**
- `useId()` フックを使用して一意なIDを生成（複数インスタンス対応）
- `fileInputRef` を追加し、リセット時にinput valueをクリア（同じファイルの再選択を可能に）

**Database Table Name Fix**
- `JdbcMemberRepository.java`: 全SQLを `member` から `members` に変更（V5マイグレーションとの整合性）
- `MemberControllerTest.kt`: テストのINSERT文も `members` テーブルに修正

**Optimistic Locking**
- `JdbcMemberRepository.save()`: バージョンチェックを追加し、更新行数を検証。競合時は `OptimisticLockingFailureException` をスロー

**Code Quality**
- `import/page.tsx`: 本番環境向けに `console.log` を削除
- `application-dev.yaml`: ロガー名前空間を `com.miometry.worklog` から `com.worklog` に修正
- `CsvExportController.java`: 未使用インポート（`MemberId`, `WorkLogStatus`）を削除
- `PerformanceBenchmarkTest.kt`: `src/test/java/` から `src/test/kotlin/` に移動

### Commit `c87d293` - Schema and code quality

**Critical fix**
- Add V9 migration to add missing 'version' column to members table (required by JdbcMemberRepository for optimistic locking)
- Update data-dev.sql to include version column in members INSERT

**Code quality improvements**
- Fix race condition in CsvImportController by deferring import cleanup to scheduled task instead of SSE callback
- Document Spring Cache limitation in evictMemberCache method

### Commit `33aab37` - Final review fixes

**Backend Security & Reliability**
- `RateLimitConfig.kt`: `cleanupIfNeeded()` を `tryConsume()` の前に移動し、allow/denyに関わらず実行するよう変更（攻撃時のメモリ圧迫防止）
- `StreamingCsvProcessor.java`: rowNumber計算を修正し、`getRecordNumber() - 1` としてCsvValidationServiceの契約（1-based, excluding header）に準拠
- `MonthlySummaryProjection.java`: パラメータ名を `fiscalYear/fiscalMonth` から `targetYear/targetMonth` に変更し、カレンダー年月であることを明確化

**Frontend Consistency & Security**
- 共通の `services/csrf.ts` モジュールを作成し、`getCsrfToken()` を共有化
- `api.ts`: `NEXT_PUBLIC_API_URL` を origin-only に正規化（csvServiceと一貫性を確保）
- `csvService.ts`: `URLSearchParams` を使用して memberId を適切にURLエンコード
- `CsvUploader.tsx`: CSV MIMEタイプのチェックを緩和し、`application/vnd.ms-excel`, `text/plain`, 空文字列などの一般的なバリアントを受け入れ

### Acknowledged (Future Improvement)
- `JdbcWorkLogRepository`: JSONクエリのパフォーマンス最適化はデータ量が増えた際に対応予定

### CODE REVIEW

File: frontend/app/services/csvService.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+export interface CsvImportResult {
+  importId: string;
+}
+
+/**
+ * Download the CSV template file.
+ */
+export async function downloadTemplate(): Promise<void> {
+  const response = await fetch(`${API_BASE_URL}/api/v1/worklog/csv/template`);
+  if (!response.ok) {
```
[the remark this answers is not in the document]
Comment (totodo713) [reply frontend/app/services/csvService.ts]: Fixed in commit `97087fd`. The base URL is now normalized using `new URL()` to extract only the protocol and host, preventing double path prefixes.

File: frontend/app/components/worklog/CsvUploader.tsx [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+          if (progress?.status === "completed") {
+            onImportComplete?.();
+          }
+        },
+      );
+    } catch (err) {
+      setError(err instanceof Error ? err.message : "Import failed");
+      setIsImporting(false);
+    }
+  }, [file, memberId, onImportComplete, progress?.status]);
```
[the remark this answers is not in the document]
Comment (totodo713) [reply frontend/app/components/worklog/CsvUploader.tsx]: Fixed in commit `eeb4b90`. Added `useEffect` cleanup to close EventSource on component unmount.

File: infra/scripts/verify-health.sh [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+if [ "$2" == "--wait" ]; then
+    wait_for_service || exit 1
+    echo ""
+fi
+
+echo "--- Health Endpoints ---"
+
+# Primary health check
+check_endpoint_json "/api/v1/health" ".status" "UP" "API Health Check"
+
```
[the remark this answers is not in the document]
Comment (totodo713) [reply infra/scripts/verify-health.sh]: Fixed in commit `eeb4b90`. Updated expected value from `UP` to `ok` to match the actual health endpoint response.

File: frontend/app/services/csvService.ts [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+): Promise<CsvImportResult> {
+  const formData = new FormData();
+  formData.append("file", file);
+  formData.append("memberId", memberId);
+
+  const response = await fetch(`${API_BASE_URL}/api/v1/worklog/csv/import`, {
+    method: "POST",
+    body: formData,
+  });
+
```
[the remark this answers is not in the document]
Comment (totodo713) [reply frontend/app/services/csvService.ts]: Fixed in commit `eeb4b90`. Added `credentials: "include"` to the fetch call for CSRF cookie handling. Also added CSRF token header support in commit `97087fd` via `api.ts`.

File: backend/src/main/java/com/worklog/api/CsvImportController.java [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+    public ResponseEntity<Map<String, String>> importCsv(
+            @RequestParam("file") MultipartFile file,
+            @RequestParam("memberId") UUID memberId) {
+        
+        if (file.isEmpty()) {
+            return ResponseEntity.badRequest()
+                    .body(Map.of("error", "File is empty"));
+        }
+
+        if (!file.getOriginalFilename().endsWith(".csv")) {
```
[the remark this answers is not in the document]
Comment (totodo713) [reply backend/src/main/java/com/worklog/api/CsvImportController.java]: Fixed in commit `eeb4b90`. Added null check for `getOriginalFilename()` before calling `.endsWith()`.

File: backend/src/main/java/com/worklog/api/CsvImportController.java [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+        progress.addEmitter(emitter);
+
+        emitter.onCompletion(() -> {
+            progress.removeEmitter(emitter);
+            activeImports.remove(importId);
+        });
+
+        emitter.onTimeout(() -> {
+            progress.removeEmitter(emitter);
+            activeImports.remove(importId);
```
[the remark this answers is not in the document]
Comment (totodo713) [reply backend/src/main/java/com/worklog/api/CsvImportController.java]: Fixed in commits `eeb4b90` and `97087fd`. Now only removing from `activeImports` when import `isFinished()`. Added TTL-based cleanup with `@Scheduled` annotation (30-minute TTL for finished imports).

File: backend/src/main/java/com/worklog/infrastructure/projection/ApprovalQueueProjection.java [hunk 7]
```diff
@@ -169,35 +170,51 @@ AND e.aggregate_id NOT IN (
...
+            
+            // Calculate intersection with requested date range
+            LocalDate effectiveStart = absenceStart.isBefore(startDate) ? startDate : absenceStart;
+            LocalDate effectiveEnd = absenceEnd.isAfter(endDate) ? endDate : absenceEnd;
+            
+            // Count days in the effective range
+            long days = effectiveStart.until(effectiveEnd).getDays() + 1;
+            
+            totalHours = totalHours.add(hoursPerDay.multiply(BigDecimal.valueOf(days)));
+        }
```
[the remark this answers is not in the document]
Comment (totodo713) [reply backend/src/main/java/com/worklog/infrastructure/projection/ApprovalQueueProjection.java]: Fixed in commit `eeb4b90`. Now using `ChronoUnit.DAYS.between(effectiveStart, effectiveEnd) + 1` for correct inclusive day counting.

File: docs/user-manual.md [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+### Understanding the Calendar
+
+| Indicator | Meaning |
+|-----------|---------|
+| Green checkmark | Entry complete with valid hours |
+| Yellow outline | Entry exists but incomplete |
+| Blue background | Weekend (Saturday/Sunday) |
+| Orange background | Holiday (no entry required) |
+| Gray background | Future dates (not yet editable) |
+| Lock icon | Month submitted for approval |
```
[the remark this answers is not in the document]
Comment (totodo713) [reply docs/user-manual.md]: Reviewed the tables in user-manual.md - they are correctly formatted with single `|` delimiters and proper header separator rows. The tables render correctly in standard Markdown. No `||` double-pipe issues found in the file.

File: docs/manager-guide.md [hunk 9]
```diff
@@ -0,0 +1,309 @@
...
+- Create proxy entries for team members when needed
+
+### Approval Targets
+
+| Metric | Target |
+|--------|--------|
+| Review completion | By 5th of following month |
+| Time per team member | < 1 minute (target: 10 members in 10 minutes) |
+| Approval accuracy | 95% on-time submissions team-wide |
+
```
[the remark this answers is not in the document]
Comment (totodo713) [reply docs/manager-guide.md]: Reviewed the tables in manager-guide.md - they are correctly formatted with single `|` delimiters and proper header separator rows (e.g., line 33-37 show `| Metric | Target |` with proper formatting). No `||` double-pipe issues found in the file.

File: infra/docker/docker-compose.prod.yml [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+      - "8080"
+
+  # Next.js Frontend
+  frontend:
+    image: miometry-frontend:${VERSION:-latest}
+    container_name: miometry-frontend
+    environment:
+      NODE_ENV: production
+      NEXT_PUBLIC_API_URL: https://${DOMAIN:-localhost}/api
+    restart: unless-stopped
```
[the remark this answers is not in the document]
Comment (totodo713) [reply infra/docker/docker-compose.prod.yml]: The double path issue was addressed in commit `97087fd` via `csvService.ts` - the base URL is now normalized using `new URL()` to extract only the protocol and host, preventing double path prefixes regardless of the `NEXT_PUBLIC_API_URL` value. The docker-compose.prod.yml can remain as-is since the frontend now correctly handles the path normalization.

File: frontend/tests/unit/components/DailyEntryForm.test.tsx [hunk 11]
```diff
@@ -34,1223 +34,1258 @@ const mockUpdateAbsence = api.absence.updateAbsence as any;
...
+
+      await waitFor(() => {
+        expect(
+          screen.getByText(/hours must be in 0\.25 increments/i),
+        ).toBeInTheDocument();
+      });
+    });
+
+    it("should reject negative hours", async () => {
+      const user = userEvent.setup();
```
[the remark this answers is not in the document]
Comment (totodo713) [reply frontend/tests/unit/components/DailyEntryForm.test.tsx]: Fixed in commit `eeb4b90`. Removed unused `user` variable from the test file.

File: frontend/tests/unit/components/DailyEntryForm.test.tsx [hunk 12]
```diff
@@ -34,1223 +34,1258 @@ const mockUpdateAbsence = api.absence.updateAbsence as any;
...
+      await waitFor(() => {
+        expect(screen.getByText("Total Daily Hours:")).toBeInTheDocument();
+        // Multiple "8.00h" elements exist (total and breakdown), so use getAllByText
+        const hourDisplays = screen.getAllByText("8.00h");
+        expect(hourDisplays.length).toBeGreaterThan(0);
+      });
+    });
+
+    it("should not allow removing the last project row", async () => {
+      const user = userEvent.setup();
```
[the remark this answers is not in the document]
Comment (totodo713) [reply frontend/tests/unit/components/DailyEntryForm.test.tsx]: Fixed in commit `eeb4b90`. Removed unused `user` variable from the test file.

File: backend/src/main/java/com/worklog/api/CsvImportController.java [hunk 13]
```diff
@@ -0,0 +1,284 @@
...
+                            // Update progress
+                            progress.update(totalRows, validRows, errorRows);
+                        }
+                );
+
+                progress.complete(result);
+            } catch (Exception e) {
+                progress.fail(e.getMessage());
+            }
+        });
```
[the remark this answers is not in the document]
Comment (totodo713) [reply backend/src/main/java/com/worklog/api/CsvImportController.java]: Fixed in commit `97087fd`. Added TTL-based cleanup with `@Scheduled` annotation (30-minute TTL for completed imports). Entries are now cleaned up even if clients never connect.

File: infra/docker/nginx/nginx.conf [hunk 14]
```diff
@@ -0,0 +1,143 @@
...
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header X-Forwarded-Proto $scheme;
+    }
+
+    # Frontend (Next.js)
+    location / {
+        limit_req zone=frontend_limit burst=100 nodelay;
+
+        proxy_pass http://frontend;
+        proxy_http_version 1.1;
```
[the remark this answers is not in the document]
Comment (totodo713) [reply infra/docker/nginx/nginx.conf]: Fixed in commit `97087fd`. Added explicit proxy locations for `/docs`, `/api-docs`, and `/static/api-docs.html` to route to the backend.

File: frontend/app/services/csvService.ts [hunk 15]
```diff
@@ -0,0 +1,148 @@
...
+      eventSource.close();
+      onComplete();
+    }
+  });
+
+  eventSource.addEventListener("error", (event) => {
+    const errorData = JSON.parse((event as MessageEvent).data);
+    onError(errorData.error || "Unknown error");
+    eventSource.close();
+  });
```
[the remark this answers is not in the document]
Comment (totodo713) [reply frontend/app/services/csvService.ts]: This is a valid concern. The EventSource error handling could be improved to guard against native connection errors. Will consider adding a check for `event instanceof MessageEvent && typeof event.data === 'string'` in a follow-up improvement.

File: infra/scripts/verify-health.sh [hunk 16]
```diff
@@ -0,0 +1,181 @@
...
+# Arguments:
+#   BASE_URL  - Base URL of the API (default: http://localhost:8080)
+#
+# Exit codes:
+#   0 - All health checks passed
+#   1 - One or more health checks failed
+#
+
+set -e
+
```
[the remark this answers is not in the document]
Comment (totodo713) [reply infra/scripts/verify-health.sh]: Fixed in commit `97087fd`. Changed from `set -e` to `set -u -o pipefail` to allow all checks to run and aggregate failures in the summary.

File: frontend/tests/unit/components/Calendar.test.tsx [hunk 17]
```diff
@@ -1,13 +1,13 @@
...
- * 
+ *
  * Tests rendering, date selection, hour display, and status indicators.
  * Task: T061 - Calendar component tests
  */
 
-import { describe, it, expect, vi, beforeEach } from "vitest";
 import { render, screen, within } from "@testing-library/react";
 import userEvent from "@testing-library/user-event";
+import { beforeEach, describe, expect, it, vi } from "vitest";
```
[the remark this answers is not in the document]
Comment (totodo713) [reply frontend/tests/unit/components/Calendar.test.tsx]: Fixed in commit `97087fd`. Removed unused `beforeEach` import from the test file.

File: infra/scripts/restore-db.sh [hunk 18]
```diff
@@ -0,0 +1,115 @@
...
+psql -h "${DB_HOST}" -p "${DB_PORT}" -U "${DB_USER}" -d postgres \
+    -c "SELECT pg_terminate_backend(pid) FROM pg_stat_activity WHERE datname = '${DB_NAME}' AND pid <> pg_backend_pid();" \
+    2>/dev/null || true
+
+# Drop and recreate database
+echo "Recreating database..."
+psql -h "${DB_HOST}" -p "${DB_PORT}" -U "${DB_USER}" -d postgres \
+    -c "DROP DATABASE IF EXISTS ${DB_NAME};"
+psql -h "${DB_HOST}" -p "${DB_PORT}" -U "${DB_USER}" -d postgres \
+    -c "CREATE DATABASE ${DB_NAME} OWNER ${DB_USER};"
```
[the remark this answers is not in the document]
Comment (totodo713) [reply infra/scripts/restore-db.sh]: Fixed in commit `e384e7f`. Changed to use psql variables with identifier quoting (`-v dbname=... -c 'DROP DATABASE IF EXISTS :"dbname";'`) to prevent SQL injection when handling DB_NAME and DB_USER.

File: frontend/app/services/csvService.ts [hunk 19]
```diff
@@ -0,0 +1,169 @@
...
+): Promise<CsvImportResult> {
+  const formData = new FormData();
+  formData.append("file", file);
+  formData.append("memberId", memberId);
+
+  const response = await fetch(`${API_BASE_URL}/api/v1/worklog/csv/import`, {
+    method: "POST",
+    body: formData,
+    credentials: "include", // Include session cookies for CSRF protection
+  });
```
[the remark this answers is not in the document]
Comment (totodo713) [reply frontend/app/services/csvService.ts]: Fixed in commit `e384e7f`. Added `getCsrfToken()` helper function that reads the XSRF-TOKEN cookie and includes it as `X-XSRF-TOKEN` header in the `importCsv()` fetch call.

File: backend/src/main/java/com/worklog/api/dto/MemberResponse.java [hunk 20]
```diff
@@ -0,0 +1,16 @@
+package com.worklog.api.dto;
+
+import java.util.List;
```
[the remark this answers is not in the document]
Comment (totodo713) [reply backend/src/main/java/com/worklog/api/dto/MemberResponse.java]: Fixed in commit `e384e7f`. Removed the unused `import java.util.List;` from MemberResponse.java.

File: frontend/app/services/api.ts [hunk 21]
```diff
@@ -12,8 +12,37 @@
  * Session timeout (30 minutes) is handled client-side via session timeout warnings.
  */
 
+import type {
+  DailyCalendarEntry,
+  MonthlyCalendarResponse,
+} from "@/types/worklog";
```
[the remark this answers is not in the document]
Comment (totodo713) [reply frontend/app/services/api.ts]: Already verified - no unused `DailyCalendarEntry` import found in current api.ts. The import was previously removed or this was a false positive.

File: frontend/app/components/worklog/MemberSelector.tsx [hunk 22]
```diff
@@ -0,0 +1,127 @@
...
+        <label
+          htmlFor="member-selector"
+          className="text-sm font-medium text-gray-700"
+        >
+          {label}
+        </label>
+      )}
+      <select
+        id="member-selector"
+        value={selectedMember?.id || ""}
```
[the remark this answers is not in the document]
Comment (totodo713) [reply frontend/app/components/worklog/MemberSelector.tsx]: Fixed in commit `59dc8cb`. Now using React `useId()` hook to generate a unique element ID instead of the hardcoded "member-selector" string, supporting multiple component instances on the same page.

File: backend/src/main/java/com/worklog/infrastructure/config/LoggingConfig.kt [hunk 23]
```diff
@@ -0,0 +1,222 @@
...
+ * - Masks sensitive headers (Authorization, Cookie, Set-Cookie)
+ * - Masks sensitive fields in JSON bodies (password, token, secret, etc.)
+ * - Configurable body logging with max length truncation
+ * - Skips health check endpoints to reduce noise
+ */
+@Component
+@Order(Ordered.HIGHEST_PRECEDENCE + 10) // Run after rate limiting
+class LoggingFilter(
+    private val properties: LoggingProperties,
+) : OncePerRequestFilter() {
```
[the remark this answers is not in the document]
Comment (totodo713) [reply backend/src/main/java/com/worklog/infrastructure/config/LoggingConfig.kt]: Fixed in commit 966530b. Updated the comment from "Run after rate limiting" to "Run early in filter chain for comprehensive request logging" to accurately describe the filter's ordering behavior.

File: backend/src/main/resources/db/migration/V5__member_table.sql [hunk 24]
```diff
@@ -0,0 +1,41 @@
...
+    organization_id UUID NOT NULL REFERENCES organization(id),
+    email VARCHAR(255) NOT NULL,
+    display_name VARCHAR(255) NOT NULL,
+    manager_id UUID REFERENCES member(id),
+    is_active BOOLEAN NOT NULL DEFAULT true,
+    version BIGINT NOT NULL DEFAULT 0,
+    created_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    updated_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    CONSTRAINT uk_member_email UNIQUE(email)
+);
```
[the remark this answers is not in the document]
Comment (totodo713) [reply backend/src/main/resources/db/migration/V5__member_table.sql]: Fixed in commit 966530b. Created V7 migration (V7__fix_member_email_unique_constraint.sql) that drops the global UNIQUE(email) constraint and replaces it with a tenant-scoped UNIQUE(tenant_id, email) constraint for proper multi-tenant support.

**Comment by totodo713** [comment]
Fixed all 15 unresolved review comments in commit `12c7786`. See individual thread replies for details.

File: frontend/app/services/csvService.ts [hunk 25]
```diff
@@ -0,0 +1,201 @@
...
+ * Spring Security sets XSRF-TOKEN cookie which we need to read
+ * and send back in X-XSRF-TOKEN header for non-GET requests.
+ */
+function getCsrfToken(): string | null {
+  if (typeof document === "undefined") {
+    return null;
+  }
+  const cookies = document.cookie.split(";");
+  for (const cookie of cookies) {
+    const [name, value] = cookie.trim().split("=");
```
[the remark this answers is not in the document]
Comment (totodo713) [reply frontend/app/services/csvService.ts]: Fixed in commit `12c7786`. Same fix applied - using `indexOf("=")` to split only on the first `=`.

File: frontend/app/services/api.ts [hunk 26]
```diff
@@ -12,8 +12,34 @@
...
+ */
+function getCsrfToken(): string | null {
+  if (typeof document === "undefined") {
+    // Server-side rendering - no cookies available
+    return null;
+  }
+
+  const cookies = document.cookie.split(";");
+  for (const cookie of cookies) {
+    const [name, value] = cookie.trim().split("=");
```
[the remark this answers is not in the document]
Comment (totodo713) [reply frontend/app/services/api.ts]: Fixed in commit `12c7786`. Same fix applied - using `indexOf("=")` to split only on the first `=`.

File: frontend/app/components/worklog/CsvUploader.tsx [hunk 27]
```diff
@@ -0,0 +1,270 @@
...
+      setIsImporting(false);
+    }
+  }, [file, memberId, onImportComplete]);
+
+  const handleReset = useCallback(() => {
+    setFile(null);
+    setProgress(null);
+    setError(null);
+    setIsImporting(false);
+  }, []);
```
[the remark this answers is not in the document]
Comment (totodo713) [reply frontend/app/components/worklog/CsvUploader.tsx]: Fixed in commit `12c7786`. Added `fileInputRef` and now clear the input value (`fileInputRef.current.value = ''`) when resetting, allowing users to re-select the same file.

File: frontend/app/worklog/import/page.tsx [hunk 28]
```diff
@@ -0,0 +1,111 @@
...
+      );
+    } finally {
+      setIsDownloading(false);
+    }
+  };
+
+  const handleImportComplete = () => {
+    // Optionally redirect to worklog page or show success message
+    console.log("Import completed successfully");
+  };
```
[the remark this answers is not in the document]
Comment (totodo713) [reply frontend/app/worklog/import/page.tsx]: Fixed in commit `12c7786`. Removed the `console.log` statement.

File: backend/src/test/kotlin/com/worklog/benchmark/PerformanceBenchmarkTest.kt [hunk 29]
```diff
@@ -0,0 +1,224 @@
+package com.worklog.benchmark
```
[the remark this answers is not in the document]
Comment (totodo713) [reply backend/src/test/kotlin/com/worklog/benchmark/PerformanceBenchmarkTest.kt]: Fixed in commit `12c7786`. Moved `PerformanceBenchmarkTest.kt` from `src/test/java/` to `src/test/kotlin/` directory.

**Review by totodo713 — COMMENTED** [review]
## Code Review Summary

### 🔴 CRITICAL: Missing `version` column in `members` table

**Problem:** `V5__member_table.sql` does not define the `version` column, but `JdbcMemberRepository.java` expects it for optimistic locking.

**Files affected:**
- `V5__member_table.sql` - Missing `version INTEGER NOT NULL DEFAULT 0`
- `data-dev.sql` - Members INSERT doesn't include `version`
- `JdbcMemberRepository.java` - Expects `version` column in all queries

**Impact:** Runtime SQL errors when accessing member endpoints. Application will fail at startup or when first member query is executed.

**Fix:** Create new migration `V9__add_member_version.sql`:
```sql
ALTER TABLE members ADD COLUMN version INTEGER NOT NULL DEFAULT 0;
```

---

### 🟡 MEDIUM: Race condition in `CsvImportController.ImportProgress` cleanup

Progress tracking cleanup has a potential race condition that could cause memory leaks under high concurrency.

---

### 🟢 LOW: Minor issues

1. `evictMemberCache` method uses `allEntries = true` instead of the `memberId` parameter
2. `CopyPreviousMonthDialog.tsx` has unused refs
3. `data-dev.sql` needs `version` column added after migration fix

---

### ✅ Good implementations

- E2E tests have comprehensive coverage
- `FiscalMonthPeriodTest.kt` has thorough edge case testing (leap years, year boundaries)
- Security: Seed data uses RFC 2606 compliant fake domains

File: backend/src/main/java/com/worklog/infrastructure/csv/StreamingCsvProcessor.java [hunk 30]
```diff
@@ -0,0 +1,162 @@
...
+
+            for (CSVRecord record : csvParser) {
+                totalRows++;
+                int rowNumber = (int) record.getRecordNumber();
+
+                // Extract values
+                String date = record.get("Date");
+                String projectCode = record.get("Project Code");
+                String hours = record.get("Hours");
+                String notes = record.size() > 3 ? record.get("Notes") : "";
```
[the remark this answers is not in the document]
Comment (totodo713) [reply backend/src/main/java/com/worklog/infrastructure/csv/StreamingCsvProcessor.java]: Fixed in commit `33aab37`. Adjusted rowNumber calculation by subtracting 1 from `getRecordNumber()` to match CsvValidationService contract (1-based, excluding header).

File: backend/src/main/java/com/worklog/infrastructure/projection/MonthlySummaryProjection.java [hunk 31]
```diff
@@ -163,45 +166,67 @@ private BigDecimal getTotalAbsenceHours(
...
         UUID memberId,
-        LocalDate startDate,
-        LocalDate endDate
+        int fiscalYear,
+        int fiscalMonth
     ) {
+        // Calculate fiscal month date range
+        LocalDate fiscalMonthStart = LocalDate.of(fiscalYear, fiscalMonth, 1).minusMonths(1).withDayOfMonth(21);
+        LocalDate fiscalMonthEnd = LocalDate.of(fiscalYear, fiscalMonth, 20);
+
```
[the remark this answers is not in the document]
Comment (totodo713) [reply backend/src/main/java/com/worklog/infrastructure/projection/MonthlySummaryProjection.java]: Fixed in commit `33aab37`. Renamed parameters from `fiscalYear/fiscalMonth` to `targetYear/targetMonth` with updated Javadoc clarifying these are calendar year/month anchors for the fiscal period (21st prev month to 20th current).

File: frontend/app/services/csvService.ts [hunk 32]
```diff
@@ -0,0 +1,205 @@
...
+  year: number,
+  month: number,
+  memberId: string,
+): Promise<void> {
+  const response = await fetch(
+    `${API_BASE_URL}/api/v1/worklog/csv/export/${year}/${month}?memberId=${memberId}`,
+    {
+      credentials: "include", // Include session cookies for cross-origin requests
+    },
+  );
```
[the remark this answers is not in the document]
Comment (totodo713) [reply frontend/app/services/csvService.ts]: Fixed in commit `33aab37`. Changed to use `URLSearchParams` for proper encoding of the memberId parameter.

File: frontend/app/components/worklog/CsvUploader.tsx [hunk 33]
```diff
@@ -0,0 +1,279 @@
...
+    const droppedFile = e.dataTransfer.files[0];
+    if (
+      droppedFile?.type === "text/csv" ||
+      droppedFile?.name.endsWith(".csv")
+    ) {
+      setFile(droppedFile);
+      setError(null);
+    } else {
+      setError("Please drop a CSV file");
+    }
```
[the remark this answers is not in the document]
Comment (totodo713) [reply frontend/app/components/worklog/CsvUploader.tsx]: Fixed in commit `33aab37`. Added `CSV_MIME_TYPES` array including common variants (`application/vnd.ms-excel`, `text/plain`, `text/x-csv`, empty string) and prioritized `.csv` extension check for reliability across browsers/OSes.

File: frontend/app/services/api.ts [hunk 34]
```diff
@@ -12,8 +12,38 @@
...
+    const eqIndex = trimmed.indexOf("=");
+    if (eqIndex === -1) continue;
+    const name = trimmed.substring(0, eqIndex);
+    const value = trimmed.substring(eqIndex + 1);
+    if (name === "XSRF-TOKEN") {
+      return decodeURIComponent(value);
+    }
+  }
+  return null;
+}
```
[the remark this answers is not in the document]
Comment (totodo713) [reply frontend/app/services/api.ts]: Fixed in commit `33aab37`. Extracted `getCsrfToken()` to a shared `services/csrf.ts` module and updated both `api.ts` and `csvService.ts` to import from it.

**Comment by totodo713** [comment]
## PR Review Comments Addressed

All 8 unresolved review comments have been addressed in commit `7f6f081`. Here's a summary of the fixes:

### 1. csvService.ts (Line 173) - URL encoding for importId
Added `encodeURIComponent(importId)` when constructing the SSE endpoint URL to prevent path injection attacks.

### 2. JdbcMemberRepository.java - Tenant scoping for findByEmail
Changed `findByEmail(String email)` to `findByEmail(TenantId tenantId, String email)` with proper tenant scoping (`WHERE tenant_id = ? AND email = ?`) to prevent cross-tenant data exposure.

### 3. RateLimitConfig.kt (Line 215) - Trusted proxy validation
Added `TrustedProxyChecker` class with CIDR support. X-Forwarded-For is now only trusted from configured trusted proxies (loopback + private ranges by default). Added `trustedProxies` config property for customization.

### 4. proxy/page.tsx - Replace hardcoded CURRENT_USER_ID
Created `useAuth` hook (`frontend/app/hooks/useAuth.ts`) with mock implementation (clearly marked as dev-only with TODO comments). Replaced all hardcoded `CURRENT_USER_ID` constants in proxy/page.tsx, worklog/page.tsx, and worklog/[date]/page.tsx with `useAuth().userId`.

### 5. globals.css (Line 57) - Clarify @theme usage
Added comment explaining that `@theme` is Tailwind CSS v4 syntax that is processed at build time by the Tailwind compiler, not raw CSS.

### 6. globals.css (Line 87) - Remove \<a\> from touch target sizing
Removed `a` from the global touch target selector to prevent breaking inline links and dense UI. Added comment recommending the use of `.touch-target` class for explicit mobile touch targets.

### 7. CsvExportController/ProjectCodeResolver (Line 77) - Graceful fallback
Changed `resolveIdToCode()` in ProjectCodeResolver to return the UUID string as a fallback for unknown project IDs instead of throwing `IllegalArgumentException`, preventing 500 errors on CSV export.

File: backend/src/main/java/com/worklog/infrastructure/config/RateLimitConfig.kt [hunk 35]
```diff
@@ -0,0 +1,187 @@
...
+     */
+    private fun getClientId(request: HttpServletRequest): String {
+        val xForwardedFor = request.getHeader("X-Forwarded-For")
+        return if (!xForwardedFor.isNullOrBlank()) {
+            // Take the first IP in the X-Forwarded-For chain (original client)
+            xForwardedFor.split(",").first().trim()
+        } else {
+            request.remoteAddr ?: "unknown"
+        }
+    }
```
[the remark this answers is not in the document]
Comment (totodo713) [reply backend/src/main/java/com/worklog/infrastructure/config/RateLimitConfig.kt]: Fixed in commit `7f6f081`. Added `TrustedProxyChecker` class with CIDR support for validating trusted proxies. X-Forwarded-For is now only trusted when the immediate client IP matches a configured trusted proxy (defaults to loopback + private ranges). Added `trustedProxies` configuration property for customization.
